src/components/app.tsx

```diff
@@ -122,29 +136,34 @@ export class App extends React.PureComponent<IProps, IState> {
           activityName={activity.name}
           singlePage={activity.layout === ActivityLayouts.SinglePage}
         />
-        <ActivityNavHeader
-          activityPages={activity.pages}
-          currentPage={currentPage}
-          fullWidth={fullWidth}
-          onPageChange={this.handleChangePage}
-          singlePage={activity.layout === ActivityLayouts.SinglePage}
-        />
-        { activity.layout === ActivityLayouts.SinglePage
-          ? this.renderSinglePageContent(activity)
-          : currentPage === 0
-            ? this.renderIntroductionContent(activity)
-            : activity.pages[currentPage - 1].is_completion
-              ? this.renderCompletionContent(activity)
-              : <ActivityPageContent
-                  enableReportButton={currentPage === activity.pages.length && enableReportButton(activity)}
-                  isFirstActivityPage={currentPage === 1}
-                  isLastActivityPage={currentPage === activity.pages.filter((page) => !page.is_hidden).length}
-                  pageNumber={currentPage}
-                  onPageChange={this.handleChangePage}
-                  page={activity.pages.filter((page) => !page.is_hidden)[currentPage - 1]}
-                  totalPreviousQuestions={totalPreviousQuestions}
-                  teacherEditionMode={this.state.teacherEditionMode}
-                />
+        { authError === ""
+          ? <ActivityNavHeader
+              activityPages={activity.pages}
+              currentPage={currentPage}
+              fullWidth={fullWidth}
+              onPageChange={this.handleChangePage}
+              singlePage={activity.layout === ActivityLayouts.SinglePage}
+            />
+          : ""
+        }
+        { authError === ""
+          ? activity.layout === ActivityLayouts.SinglePage
+            ? this.renderSinglePageContent(activity)
+            : currentPage === 0
+              ? this.renderIntroductionContent(activity)
+              : activity.pages[currentPage - 1].is_completion
+                ? this.renderCompletionContent(activity)
+                : <ActivityPageContent
+                    enableReportButton={currentPage === activity.pages.length && enableReportButton(activity)}
+                    isFirstActivityPage={currentPage === 1}
+                    isLastActivityPage={currentPage === activity.pages.filter((page) => !page.is_hidden).length}
+                    pageNumber={currentPage}
+                    onPageChange={this.handleChangePage}
+                    page={activity.pages.filter((page) => !page.is_hidden)[currentPage - 1]}
+                    totalPreviousQuestions={totalPreviousQuestions}
+                    teacherEditionMode={this.state.teacherEditionMode}
+                  />
+          : this.renderAuthError()
```

Comment:
You could even collapse this and the above into a single check for `authError === ""` where you would conditionally render both the `ActivityNavHeader` and the contents or the `<AuthError />`.  You would need to wrap the `ActivityNavHeader` and the contents  inside a React fragment (which could all live in a new function).  Just a thought to reduce the check for the failed authentication to a single code location.